Sources/Redis/RedisProvider.swift

```diff
@@ -3,27 +3,23 @@ import Vapor
 public struct RedisProvider: Provider {
     public init() { }
 
-    public func register(_ s: inout Services) {
-        s.register(RedisConnectionSource.self) { c in
-            return try RedisConnectionSource(config: c.make(), eventLoop: c.eventLoop)
+    public func register(_ app: Application) {
+        app.register(RedisConnectionSource.self) { app in
+            return RedisConnectionSource(configuration: app.make())
         }
 
-        s.register(ConnectionPoolConfig.self) { c in
-            return .init()
+        app.register(singleton: ConnectionPool<RedisConnectionSource>.self, boot: { app in
+            return ConnectionPool(configuration: app.make(), source: app.make(), logger: app.make(Logger.self), on: app.make())
+        }) { pool in
```

Comment:
minor nitpick - since there's two closures here, lets pass both as explicit arguments with their labels, rather than the trailing closure syntax.

Reply:
Good catch! Feel free to tidy up anything else you see fit or ask for other changes, as I'm not sure of your preferred style of indentation etc.